popup.js

```diff
@@ -1,38 +1,81 @@
-let active = document.getElementById('active');
+const active = document.getElementById('active');
+const activeLabel = document.getElementById('on-switch-label');
+const intervalSlider = document.getElementById('reload-int-input');
+const intervalDisplay = document.getElementById('reload-int-display');
+const settingsContainer = document.getElementById('settings-container');
+const seconds = document.getElementById('sec');
+const minutes = document.getElementById('min');
 
+// Maintain checked input box if this window has active reload
 chrome.tabs.query({ currentWindow: true, active: true }, function (tabs) {
   let tab = tabs[0];
   chrome.storage.local.get(tab.id.toString(), (items) => {
-      if(items[tab.id]){
-          console.log("found!");
-          active.checked = true;
-          
+    // Load previous user input settings
+    if (items[tab.id]) {
+      // Load unit check box
+      if (items[tab.id].unit === 'min') {
+        minutes.checked = true;
+      } else {
+        seconds.checked = true;
       }
+
+      // Load Interval display
+      intervalDisplay.innerHTML = items[tab.id].input;
+      intervalSlider.value = items[tab.id].input;
+    }
+
+    // Check off checkbox if currently reloading
+    if(items[tab.id] && items[tab.id].active){
+        active.checked = true;
+    }
   });
 });
 
+// Save current window tab to Chrome storage
 active.onchange = function(event) {
-  if(event.target.checked){
-    let interval = 10000;
-    
+  // Save reload interval parameter to Chrome storage
+  let numInput = parseInt(intervalSlider.value);
+  // Toggle to minutes if the option was selected
+  let unitInput = minutes.checked ? 'min' : 'sec';
+
+  // Turn ON
+  if (event.target.checked) {
     chrome.tabs.query({ currentWindow: true, active: true }, function (tabs) {
-      console.log(tabs[0]);
       let tab = tabs[0];
-      let tostore = {};
-      tostore[tab.id] = true;
-      chrome.storage.local.set(tostore);
+      let storedObj = {};
+      storedObj[tab.id] = {
+        active: true,
+        input: numInput,
+        unit: unitInput
+      };
+      chrome.storage.local.set(storedObj);
     });
+
+    activeLabel.innerHTML = 'ON';
```

Comment:
Also do you think it will be better if we store Label strings "ON"/"OFF" in some variable so that we do not have to manually update everywhere if we need to?

Reply:
Is this to just optimize the performance slightly to call a reference from a stored variable rather than writing a new string every time? Or is there a particular way you'd like me to update the label string?

Reply:
No, nothing about performance (at least from intention of my comment).
Take a hypothetical situation where we want to put "Active" instead of "ON". To make this change, we will have to replace all the "ON" strings with "Active" and if we miss this replacement at a single place, the labels will be inconsistent. In some cases the label will show ON and "Active" in other cases. 
If we have a variable called `const onLabel = "ON";` and we use this variable everywhere, next time when intend to change the label, we just change this variable and the all places where this variable is referenced will automatically get the update! 

I hope you understood this. This is not critical right now, but it is a good practise.

Reply:
Thanks for the explanation, it makes a lot of sense.